// Copyright 2019 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// A protocol buffer encoding one or several classifications in an image. A
// classification is defined by label and corresponding score, representing the
// classifier confidence for the label.

syntax = "proto2";

package mediapipe;

message Classification {
  // The index of the class in the corresponding label map.
  optional int32 index = 1;
  // The probability score for this class.
  optional float score = 2;
  // Label or name of the class.
  optional string label = 3;
}

// Group of Classification protos.
message ClassificationList {
  repeated Classification classification = 1;
}
